include/seastar/coroutine/exception.hh

```diff
@@ -68,7 +68,11 @@ namespace coroutine {
 /// ```
 struct exception {
     std::exception_ptr eptr;
-    explicit exception(std::exception_ptr eptr) : eptr(std::move(eptr)) {}
+
+    explicit exception(std::exception_ptr&& eptr) noexcept : eptr(std::move(eptr)) {}
+
+    template <typename E>
+    explicit exception(E&& e) noexcept : eptr(std::make_exception_ptr(std::forward<E>(e))) {}
```

Comment:
Obligatory bikeshed: should this be unconditionally noexcept? `std::make_exception_ptr` takes its argument by value, so we need to move/copy `e` which can throw. I guess crashing might be the most sensible option here, just wanted to point this out in case this was not intentional.

Reply:
This is in line with other places we don't let low-level exception handling throw.

Reply:
Ref: https://github.com/scylladb/seastar/blob/1424d34c93846c8e23d60e7550f39d621643ac8e/include/seastar/core/future.hh#L2086-L2089

Reply:
I see, thanks.